core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py

```diff
@@ -53,38 +55,64 @@ def generate(self):
 
         if self.is_upgrade_run:
             # For upgrade at this point we don't need any of other roles then
-            # common, upgrade, repository and image_registry.
+            # common, upgrade, repository, image_registry and haproxy.
             # - commmon is already provisioned from the cluster model constructed from the inventory.
             # - upgrade should not require any additional config
             # roles in the list below are provisioned for upgrade from defaults
-            enabled_roles = ['repository', 'image_registry']
+            enabled_roles = ['repository', 'image_registry', 'haproxy']
         else:
             enabled_roles = self.inventory_creator.get_enabled_roles()
 
         for role in enabled_roles:
-            document = select_first(self.config_docs, lambda x: x.kind == 'configuration/'+to_feature_name(role))
+            kind = 'configuration/' + to_feature_name(role)
 
+            document = select_first(self.config_docs, lambda x: x.kind == kind)
             if document is None:
                 self.logger.warn('No config document for enabled role: ' + role)
                 continue
 
             document.specification['provider'] = self.cluster_model.provider
             self.write_role_vars(ansible_dir, role, document)
+            self.write_role_manifest_vars(ansible_dir, role, kind)
 
         self.populate_group_vars(ansible_dir)
 
-    def write_role_vars(self, ansible_dir, role, document):
+    def write_role_vars(self, ansible_dir, role, document, vars_file_name='main.yml'):
         vars_dir = os.path.join(ansible_dir, 'roles', to_role_name(role), 'vars')
         if not os.path.exists(vars_dir):
             os.makedirs(vars_dir)
 
-        vars_file_name = 'main.yml'
         vars_file_path = os.path.join(vars_dir, vars_file_name)
 
         with open(vars_file_path, 'w') as stream:
             dump(document, stream)
 
-        self.roles_with_generated_vars.append(to_role_name(role))
+        if vars_file_name == 'main.yml':
+            self.roles_with_generated_vars.append(to_role_name(role))
+
+    def write_role_manifest_vars(self, ansible_dir, role, kind):
+        enabled_kinds = {"configuration/haproxy"}
```

Comment:
Why dict is used here and not tuple for example?

Reply:
Could you explain why do we need both `enabled_roles` and `enabled_kinds`?

Reply:
> Why dict is used here and not tuple for example?

This is a set(), not a dict(). In python {} have double meaning, depending on syntax of the literal.

Reply:
> Could you explain why do we need both `enabled_roles` and `enabled_kinds`?

This is just an optimization. This is a new thing never done before, so after some discussion with @to-bar we decided to narrow the scope of this operation. Default behavior is still to use default values.

Reply:
> > Why dict is used here and not tuple for example?
> 
> This is a set(), not a dict(). In python {} have double meaning, depending on syntax of the literal.

Agree, it's my inattention